File: 백준/Silver/1926. 그림/그림.swift
var arr:[[Int]] = [[]]
var visited:[[Bool]] = [[]]
let dx = [0,0,1,-1,1,1,-1,-1]
let dy = [1,-1,0,0,1,-1,1,-1]
var cnt = 0
var m = 0
var n = 0

var max = 0
var size = 0

var input = readLine()!.split(separator: " ").map{Int(String($0))!}
n = input[0]
m = input[1]

arr = Array(repeating: Array(repeating: 0, count: m), count: n)
visited = Array(repeating: Array(repeating: false, count: m), count: n)

for i in 0...n-1{
    var aa = readLine()!.split(separator: " ").map{Int(String($0))!}
    for j in 0...m-1{
        arr[i][j] = aa[j]
    }
}
for i in 0...n-1{
    for j in 0...m-1{
        size = 1
        if(arr[i][j] == 1 && !visited[i][j]){
            dfs(i,j)
            cnt += 1
            if(max < size){
                max = size
            }
        }
    }
}

print(cnt)
print(max)


func dfs(_ x:Int,_ y:Int){
    visited[x][y] = true
    for i in 0...3{
        let dx = x + dx[i]
        let dy = y + dy[i]
        if(dx >= 0 && dx < n && dy >= 0 && dy < m && !visited[dx][dy]){
            if(arr[dx][dy] == 1){
                dfs(dx, dy)
                size += 1
            }
            
        }
    }
}

File: 백준/Silver/1926. 그림/그림.java
import java.io.*;
import java.security.KeyStore;
import java.util.*;

public class Main {
    public static int n;
    public static int m;
    public static int area;
    public static int cnt=0;
    public static int [][] arr;
    public static boolean [][] visited;
    public static void dfs(int x, int y){
        if(arr[x][y]==1){
            visited[x][y]=true;
            area++;
            if(x+1<n && arr[x+1][y]==1 && !visited[x+1][y]){
                dfs(x+1,y);
            }
            if(x-1>=0 && arr[x-1][y]==1 && !visited[x-1][y]){
                dfs(x-1,y);
            }
            if(y+1<m && arr[x][y+1]==1 && !visited[x][y+1]){
                dfs(x,y+1);
            }
            if(y-1>=0 && arr[x][y-1]==1 && !visited[x][y-1]){
                dfs(x,y-1);
            }
        }
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String [] A = br.readLine().split(" ");
        n = Integer.parseInt(A[0]);
        m = Integer.parseInt(A[1]);
        arr = new int[n][m];
        visited = new boolean[n][m];
        ArrayList<Integer>list = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String [] B = br.readLine().split(" ");
            for (int j = 0; j < m; j++) {
                arr[i][j]= Integer.parseInt(B[j]);
            }
        }
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if(!visited[i][j]){
                    area=0;
                    dfs(i,j);
                    if(area!=0) {
                        list.add(area);
                        cnt++;
                    }
                }
            }
        }
        Collections.sort(list);
        System.out.println(cnt);
        if(list.isEmpty()) {
            System.out.println(0);
        }
        else {
            System.out.println(list.get(list.size() - 1));
        }
//        System.out.println();
//        for (int i = 0; i < n; i++) {
//            for (int j = 0; j < m; j++) {
//                System.out.print(arr[i][j]+" ");
//            }
//            System.out.println();
//        }










        bw.flush();
        bw.close();
        br.close();
    }
}